### ISSUE TITLE
Suggestion: Analyzer for Tagged comments

### ISSUE THREAD

**Issue by fitdev** [wmjordan/Codist#174]
It would be nice to be able to have some user-defined comment tags participate in code analysis, such that certain tagged comments can produce either warnings or errors.

So, at the basic level, a 3rd dropdown is needed on *tags* section in the *Syntax Highlight Configurations* called Analyzer (or something similar) with values:
- None: Does not participate in code analysis at all
- Warning: If tag is present a compiler warning is issued
- Error: If tag is present, it's a compile time error

Maybe for Warning and Error add ability to define custom, per-tag analyzer IDs.

### A More Advanced Version:

To make this even more useful, whether or not a Warning/Error is issued can be made to depend on not just on the comment tag itself (string value), but on additional conditions:
- Project configuration / compile time #constant
- Date - to enable reminders / todo-like functionality

To implement it as an Analyzer in an extension, you can look into how they do analyzers in [Roslynator](https://github.com/JosefPihrt/Roslynator)

**Comment by fitdev** [comment]
**Additional Ideas about Tagged Comments:**

1. Ability to list / display such tagged comments in the NaviBar's dropdowns:

- Current File Node (first item)
- Respective place within the namespace / type drodowns

2. Ability to display these in Super QuickInfos (perhaps in a similar place where Analyzer comment is displayed)

**Comment by fitdev** [comment]
Another possibility is to perhaps somehow integrate it with VS's task list.

**Comment by wmjordan** [comment]
Aren't they already in the Task List?

<img width="1028" alt="image" src="https://user-images.githubusercontent.com/12182730/178624086-c38f941d-f0e6-495a-b8fa-663c24ec1b11.png">

**Comment by fitdev** [comment]
You are right! Sorry about that. I guess I forgot to keep the 2 lists in sync (I have more comment tags defined in Codist than I have defined in VS's task list).
